crates/uv-distribution-types/src/index.rs

```diff
@@ -349,6 +375,25 @@ impl<'a> From<&'a IndexUrl> for IndexMetadataRef<'a> {
     }
 }
 
+/// Validate the provided error codes.
+fn validate_error_codes<'de, D>(deserializer: D) -> Result<Vec<u16>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let status_codes = Option::<Vec<u16>>::deserialize(deserializer)?;
+    if let Some(codes) = status_codes {
+        for code in &codes {
+            if StatusCode::from_u16(*code).is_err() {
+                return Err(serde::de::Error::custom(format!(
+                    "{code} is not a valid HTTP status code"
+                )));
+            }
+        }
+        return Ok(codes);
```

Comment:
Why not just store `StatusCode` directly instead of the u16?

Reply:
`StatusCode` doesn't implement `Serialize`

Reply:
Why does it need to implement `Serialize`?

Reply:
Both `Serialize` and `Deserialize` are derived on `Index` and `StatusCode` doesn't implement either. We could do `#[serde(skip)]` but that seemed incorrect to me.

Reply:
Stepping up a level, why is `Index` serializable? For writing it to the TOML file? If that's it, why _would_ we need this field to be serializable? If truly necessary, why not use a wrapper type to implement serializability?

Reply:
If we don’t need it, I can use #[serde(skip)]

Reply:
It's going to be more complicated than that. We can't use skip when we need a `deserialize_with` to parse the numbers in the TOML file. I'll need to implement a wrapper type around `StatusCode` for deriving `schemars::JsonSchema` as well as `De/Serialize` if we want to do it this way.

Reply:
It seems nice for type safety throughout and straightforward to implement? I don't feel super strongly, but the current approach of validating at the CLI level, but retaining the u16, then revalidating later seems quite weird.

Reply:
Yeah that’s why originally it was validated once on ingest with an expect down the line. But I can also create the wrapper